std-bits/table/src/main/java/org/enso/table/operations/GroupingOrderingVisitor.java

```diff
@@ -16,58 +16,78 @@
 import org.enso.table.problems.ProblemAggregator;
 import org.enso.table.util.ConstantList;
 
-abstract class RunningLooper<T> {
-
-  // implement this method in subclasses to control the order you want to loop over the data
-  public abstract void loopImpl(RunningStatistic<T> runningStatistic, long numRows);
-
-  public static <T> void loop(
+/**
+ * Abstract class GroupingOrderingVisitor
+ *
+ * <p>Overview: This class provides a mechanism for visiting rows of data based on grouping and
+ * ordering criteria. Usage : GroupingOrderingVisitor.visit( groupingColumns, orderingColumns,
+ * directions, problemAggregator, rowVisitorFactory, sourceColumn.getSize());
+ *
+ * <p>Static Method: - static void visit( Column[] groupingColumns, Column[] orderingColumns, int[]
+ * directions, ProblemAggregator problemAggregator, RowVisitorFactory visitor, long numRows)
+ * Description: For each group: will call getNewRowVisitor() Then for each row in that group will
```

Comment:
I feel like the newlines in this comment got a bit mangled by the formatter?

Maybe we could try being closer to the Javadoc format with `@param` etc. - I guess the formatter may play better then.